src/MoBi.Presentation/Presenter/HierarchicalStructurePresenter.cs

```diff
@@ -111,7 +111,10 @@ public virtual void Select(ObjectBaseDTO objectBaseDTO)
 
       private void raiseEntitySelectedEvent(ObjectBaseDTO objectBaseDTO)
       {
-         _context.PublishEvent(new EntitySelectedEvent(objectBaseDTO.ObjectBase, this));
+         // First and second neighbor node selections should not trigger an EntitySelectedEvent
+         // because the are not ObjectBase
+         if(objectBaseDTO.ObjectBase != null)
+            _context.PublishEvent(new EntitySelectedEvent(objectBaseDTO.ObjectBase, this));
```

Comment:
Probably better to not publish the event since no other presenters can handle it I guess